eslint.config.mjs

```diff
@@ -0,0 +1,74 @@
+// @ts-check
+import { FlatCompat } from '@eslint/eslintrc';
+import { fixupConfigRules, fixupPluginRules } from '@eslint/compat';
+import eslint from '@eslint/js';
+import eslintConfigPrettier from 'eslint-config-prettier';
+import reactPlugin from 'eslint-plugin-react';
+import simpleImportSortPlugin from 'eslint-plugin-simple-import-sort';
+import globals from 'globals';
+import tsEslint from 'typescript-eslint';
+import { fileURLToPath } from 'url';
+
+const tsconfigRootDir = fileURLToPath(new URL('.', import.meta.url)),
+  flatCompat = new FlatCompat();
+
+export default tsEslint.config(
+  // register all of the plugins up-front
+  {
+    plugins: {
+      '@typescript-eslint': tsEslint.plugin,
+      // @ts-expect-error https://github.com/jsx-eslint/eslint-plugin-react/issues/3699
+      react: fixupPluginRules(reactPlugin),
+      'simple-import-sort': simpleImportSortPlugin,
+    },
+  },
+  {
+    // config with just ignores is the replacement for `.eslintignore`
+    ignores: ['**/node_modules/**', '**/public/**', '**/.next/**'],
+  },
+
+  // extends ...
+  eslint.configs.recommended,
+  ...tsEslint.configs.recommended,
+  ...fixupConfigRules(flatCompat.extends('next/core-web-vitals')),
```

Comment:
This resolution throws an error:
```text
ESLint: 9.13.0

Error: Cannot read config file: ~/KaiYuanShe/kaiyuanshe.github.io/node_modules/.pnpm/eslint-config-next@15.0.0_eslint@9.13.0_typescript@5.6.3/node_modules/eslint-config-next/index.js
Error: Failed to patch ESLint because the calling module was not recognized.
```